### ISSUE TITLE
--ref/-r flag for odo create doesn't work

### ISSUE THREAD

**Issue by anandrkskd** [openshift/odo#3498]
/kind bug

## What versions of software are you using?

**Operating System:** Supported

**Output of `odo version`:** master

## How did you run odo exactly?
```
1. make valid small change in server.js and made commit
2. odo create nodejs -r _commit/branch_
3. odo push
3. check for changes manually inside pod or from endpoint according to change 
```

## Actual behavior
odo is creating components with current branch HEAD in case of commit as reference and for branch as reference its considering current branch.

## Expected behavior
odo should create componentes according to the reference passed in `odo create -r commit/branch` .

## Any logs, error output, etc?

N/A

**Comment by kadel** [comment]
the `-r` flag is only valid when using `--git` components. 
We need to update help message to make this clear`

**Comment by girishramnani** [comment]
I can work on this

**Comment by girishramnani** [comment]
opened a PR for this https://github.com/openshift/odo/pull/3525

**Comment by kadel** [comment]
> I can work on this

this is really low priority. We have a lot more pressing issues in "For Consideration" pile or labeled with required-for-v2

**Comment by kadel** [comment]
fixed in #3525 
/close
